pandas/io/excel/_base.py

```diff
@@ -532,17 +539,20 @@ def parse(
         ret_dict = False
 
         # Keep sheetname to maintain backwards compatibility.
+        sheets: list[int] | list[str]
         if isinstance(sheet_name, list):
             sheets = sheet_name
             ret_dict = True
         elif sheet_name is None:
             sheets = self.sheet_names
             ret_dict = True
+        elif isinstance(sheet_name, str):
```

Comment:
I guess this is for mypy, an alternative could be a cast in the else: `sheets = cast("list[int] | list[str]", [sheet_name])`

Reply:
I would prefer this, because we had some performance problems with cast, but not a strong opinion.

Reply:
I think, the issue is that the arguments to cast need to be evaluated. This can be slow if we have Unions/Generic types, if we "just" have a string, we should be fine (not sure whether this is the case).

Reply:
Looked into this again. If we want to use cast we have to declare 

```
sheets: list[str] | list[int] | list[int | str]
```

or
```
sheets: Any
```
before doing the assignment. I would prefer doing it with the isinstance, is a bit more cleare to me

Reply:
okay